src/super_gradients/training/transforms/keypoint_transforms.py

```diff
@@ -0,0 +1,302 @@
+import random
+from abc import abstractmethod
+from typing import Tuple, List, Iterable
+
+import cv2
+import numpy as np
+from torchvision.transforms import functional as F
+
+__all__ = [
+    "KeypointsNormalize",
+    "KeypointsToTensor",
+    "KeypointsPadIfNeeded",
+    "KeypointsLongestMaxSize",
+    "KeypointTransform",
+    "KeypointsCompose",
+    "KeypointsRandomHorizontalFlip",
+    "KeypointsRandomAffineTransform",
+    "KeypointsRandomVerticalFlip",
+]
+
+
+class KeypointTransform(object):
+    @abstractmethod
+    def __call__(self, image: np.ndarray, mask: np.ndarray, joints: np.ndarray) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """
+
+        :param image: [H,W,3]
+        :param mask: [H,W]
+        :param joints: [Instances,Joints,3]
+        :return: (image, mask, joints)
+        """
+        raise NotImplementedError
+
+
+class KeypointsCompose(KeypointTransform):
+    def __init__(self, transforms: List[KeypointTransform]):
+        self.transforms = transforms
+
+    def __call__(self, image, mask, joints):
+        for t in self.transforms:
+            image, mask, joints = t(image, mask, joints)
+        return image, mask, joints
+
+    def __repr__(self):
+        format_string = self.__class__.__name__ + "("
+        for t in self.transforms:
+            format_string += "\n"
+            format_string += "    {0}".format(t)
+        format_string += "\n)"
+        return format_string
+
+
+class KeypointsToTensor(KeypointTransform):
+    def __call__(self, image: np.ndarray, mask: np.ndarray, joints: np.ndarray):
+        return F.to_tensor(image), mask, joints
+
+
+class KeypointsNormalize(KeypointTransform):
```

Comment:
We can have a base class ImageOnlyTransform class for photometric augmentations. But checking whether a given augmentation is Image-only a compose class would then pass only image in it.

Reply:
mmmm but then again we would want all of the torch vision transforms to be included here, so that kind of defeats the purpose unless we define the torch vision transforms type to be included in ImageOnlyTransform type.

Reply:
I agree that suggested change is a movement in the right direction. And perhaps we can schedule making it somewhere in upcoming sprints. But I feel it would affect transforms for other tasks, thus diluting the scope of change of this particular PR. Therefore if you're not strongly disagree, let's move the requested change to another ticket.